node/impl/eth/gas.go

```diff
@@ -75,33 +76,38 @@ func (e *ethGas) EthGasPrice(ctx context.Context) (ethtypes.EthBigInt, error) {
 	return ethtypes.EthBigInt(gasPrice), nil
 }
 
-func (e *ethGas) EthFeeHistory(ctx context.Context, p jsonrpc.RawParams) (ethtypes.EthFeeHistory, error) {
+func (e *ethGas) EthFeeHistory(ctx context.Context, p jsonrpc.RawParams) (*ethtypes.EthFeeHistory, error) {
 	params, err := jsonrpc.DecodeParams[ethtypes.EthFeeHistoryParams](p)
 	if err != nil {
-		return ethtypes.EthFeeHistory{}, xerrors.Errorf("decoding params: %w", err)
+		return nil, xerrors.Errorf("decoding params: %w", err)
 	}
 	if params.BlkCount > 1024 {
-		return ethtypes.EthFeeHistory{}, xerrors.New("block count should be smaller than 1024")
+		return nil, xerrors.New("block count should be smaller than 1024")
 	}
 	rewardPercentiles := make([]float64, 0)
 	if params.RewardPercentiles != nil {
 		if len(*params.RewardPercentiles) > maxEthFeeHistoryRewardPercentiles {
-			return ethtypes.EthFeeHistory{}, xerrors.New("length of the reward percentile array cannot be greater than 100")
+			return nil, xerrors.New("length of the reward percentile array cannot be greater than 100")
 		}
 		rewardPercentiles = append(rewardPercentiles, *params.RewardPercentiles...)
 	}
 	for i, rp := range rewardPercentiles {
 		if rp < 0 || rp > 100 {
-			return ethtypes.EthFeeHistory{}, xerrors.Errorf("invalid reward percentile: %f should be between 0 and 100", rp)
+			return nil, xerrors.Errorf("invalid reward percentile: %f should be between 0 and 100", rp)
 		}
 		if i > 0 && rp < rewardPercentiles[i-1] {
-			return ethtypes.EthFeeHistory{}, xerrors.Errorf("invalid reward percentile: %f should be larger than %f", rp, rewardPercentiles[i-1])
+			return nil, xerrors.Errorf("invalid reward percentile: %f should be larger than %f", rp, rewardPercentiles[i-1])
 		}
 	}
 
 	ts, err := e.tipsetResolver.GetTipsetByBlockNumber(ctx, params.NewestBlkNum, false)
 	if err != nil {
-		return ethtypes.EthFeeHistory{}, err // don't wrap, to preserve ErrNullRound
+		// According to go-ethereum patterns, "not found" errors should lead to (nil, nil).
+		// Other errors should result in (nil, err).
+		if errors.Is(err, &api.ErrNullRound{}) || ipld.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
```

Comment:
nice, do we happen to have a test for this?

Reply:
actually I see this pattern is the same one you're using in most places - but I do think basic tests for these would be good, maybe in eth_api_test.go, or fevm_test.go even, let me know if you need guidance in setting something up, it should be relatively straightforward once you establish a pattern